Fix scan_handlers.rs Using Non-Existent state.pool Field Instead of state.db

## Bug Description
All three handler functions in `backend/api/src/scan_handlers.rs` reference `state.pool`, but the `AppState` struct in `backend/api/src/state.rs` defines the database pool field as `db`, not `pool`.

## Affected Code
`backend/api/src/scan_handlers.rs`:
```rust
// Line 16 — ingest_cves handler
match scanner_service::sync_cves(&state.pool, payload).await {

// Line 33 — scan_contract handler
match scanner_service::perform_scan(&state.pool, contract_id, payload).await {

// Line 46 — get_scan_report handler
match scanner_service::get_history(&state.pool, contract_id).await {
```

`backend/api/src/state.rs`:
```rust
pub struct AppState {
    pub db: PgPool,       // ← field is named 'db'
    pub started_at: Instant,
    pub cache: Arc<CacheLayer>,
    pub registry: Registry,
    pub is_shutting_down: Arc<AtomicBool>,
}
```

## Steps to Reproduce
1. Uncomment `mod scan_handlers;` in `main.rs` and add scan routes
2. Run `cargo build -p api`
3. Build fails: `error[E0609]: no field 'pool' on type 'AppState'`

## Expected Behavior
The scan handlers should compile and correctly access the database pool.

## Proposed Fix
- Replace all `state.pool` references with `state.db` in `scan_handlers.rs` (lines 16, 33, 46)
- Verify that `scanner_service` functions accept `&PgPool` as their first parameter
- Add scan routes to the router and write integration tests
- Consider adding a compile-time check (CI step) that builds all modules, even those currently commented out

--- a/backend/api/src/scan_handlers.rs
+++ b/backend/api/src/scan_handlers.rs
@@ -13,7 +13,7 @@ pub async fn ingest_cves(
     State(state): State<AppState>,
     Json(payload): Json<Vec<VulnerabilityPayload>>,
 ) -> impl IntoResponse {
-    match scanner_service::sync_cves(&state.pool, payload).await {
+    match scanner_service::sync_cves(&state.db, payload).await {
         Ok(count) => {
             let msg = format!("Ingested {} CVEs successfully", count);
             (StatusCode::OK, Json(msg)).into_response()
@@ -30,7 +30,7 @@ pub async fn scan_contract(
     Path(contract_id): Path<Uuid>,
     Json(payload): Json<ScanRequest>,
 ) -> impl IntoResponse {
-    match scanner_service::perform_scan(&state.pool, contract_id, payload).await {
+    match scanner_service::perform_scan(&state.db, contract_id, payload).await {
         Ok(report) => (StatusCode::OK, Json(report)).into_response(),
         Err(e) => {
             let err = format!("Failed to run contract scan: {}", e);
@@ -43,7 +43,7 @@ pub async fn get_scan_report(
     State(state): State<AppState>,
     Path(contract_id): Path<Uuid>,
 ) -> impl IntoResponse {
-    match scanner_service::get_history(&state.pool, contract_id).await {
+    match scanner_service::get_history(&state.db, contract_id).await {
         Ok(report) => (StatusCode::OK, Json(report)).into_response(),
         Err(e) => {
             let err = format!("Failed to retrieve scan history: {}", e);

